### PR TITLE
Fixes for eth_getBlockByNumber

### LINKED ISSUE #1675: Z2-Converter API Mainnet : eth_getBlockByNumber
### Issue Summary:
Calling `eth_getBlockByNumber` on the ZQ1 and ZQ2 has the following diff.
Here are the URLs for ZQ1 and ZQ2.
ZQ2 : http://34.124.218.159:4201/
ZQ1 : http://34.124.185.137:4201/
### Result:
**Missing keys:**
```
Keys present in ZQ1 but missing in ZQ2:
  root['result']['version']

Keys present in ZQ2 but missing in ZQ1:
  root['result']['view']
  root['result']['mixHash']
  root['result']['quorumCertificate']

```
**Different Keys:**
```
Values that differ between ZQ2 and ZQ1:
  root['result']['difficulty']:
    ZQ2: 0x0
    ZQ1: 0x68
  root['result']['gasLimit']:
    ZQ2: 0x501bd00
    ZQ1: 0x1045a640
  root['result']['hash']:
    ZQ2: 0x7c7c754408e16f11b93921ce9ee6aa704cd3189e06d9b2e141efbd3ae0f93b35
    ZQ1: 0x424da7e159b0a4504ce307952b285b7eb6ed2693df41cc81a8e96862f58c4b9b
  root['result']['miner']:
    ZQ2: 0x0000000000000000000000000000000000000001
    ZQ1: 0x14777730a5a767545625eed65679c03ec29b1c9e
  root['result']['parentHash']:
    ZQ2: 0xe29d9c1aefde7728284a240584045b33e67f54cd061611401d0313f910b4f038
    ZQ1: 0x4ab87748ca5aaf0dd451c01f877ff92b4dd918a8c31fdf95872ff90368d76a62
  root['result']['receiptsRoot']:
    ZQ2: 0x7f1d41f360b4365aa0292bfc8add72dd38bba019edea6ec1667a8db403412561
    ZQ1: 0x0000000000000000000000000000000000000000000000000000000000000000
  root['result']['sha3Uncles']:
    ZQ2: 0x0000000000000000000000000000000000000000000000000000000000000000
    ZQ1: 0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347
  root['result']['size']:
    ZQ2: 0x0
    ZQ1: 0x78d
  root['result']['stateRoot']:
    ZQ2: 0x0297829e719795a14b56f5dcb875b7546c6a4d9e0b10190aecef256319cd3335
    ZQ1: 0x3a600ca11a0afb3ec7cbd3467e4d7d65ab7e6ba9514bf53e9d549e207cf07e6d
  root['result']['transactionsRoot']:
    ZQ2: 0x9a62940175ec43145f56c87b29688f98f61b160fa821d3303aeb31c94e382cf8
    ZQ1: 0x0000000000000000000000000000000000000000000000000000000000000000

```
### Observations:
1) The version field is missing in ZQ2.
2) New fields such as view, mixHash, and quorumCertificate have been introduced in ZQ2.
3) Significant differences exist between various data fields in ZQ1 and ZQ2.

###  Conclusion:
If the above differences are expected and align with the migration to ZQ2, no further action is required. However, if these differences are unexpected, they may indicate issues that must be addressed.

### PR DESCRIPTION
**Pull request by maxconway**
- Added a version field
- Made the size field real

This is intended to close #1675.
There are still value differences remaining, I'm going to assume they're fine unless anyone knows otherwise

### CODE REVIEW

File: zilliqa/src/message.rs [hunk 1]
```diff
@@ -696,6 +707,34 @@ impl Block {
     pub fn gas_limit(&self) -> EvmGas {
         self.header.gas_limit
     }
+    pub fn size(&self) -> u64 {
+        let mut size = 0;
+
+        // Size of BlockHeader
+        size += std::mem::size_of_val(&self.header.view) as u64;
```
[the remark this answers is not in the document]
Comment (maxconway) [reply zilliqa/src/message.rs]: 1. Yep, good call
2. I've actually changed it to keeping to usize internally, for consistency with typical `.size()` methods, then converting to u64 just at the point of returning from the API. It shouldn't make a difference but it feels cleaner this way.
3. I've also converted some of the size functions from creating an approach of starting with `let mut size` and then `+=` to it to instead just being big sums. Should compile to the same thing but it seems a bit cleaner to me

File: zilliqa/src/api/types/eth.rs [hunk 2]
```diff
@@ -101,13 +101,14 @@ pub struct Block {
     pub quorum_certificate: QuorumCertificate,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub aggregate_quorum_certificate: Option<AggregateQc>,
+    pub version: u64,
```
Comment (JamesHinshelwood) [inline zilliqa/src/api/types/eth.rs:104]: Why did we add this field? It isn't documented in any of the `eth_getBlockByNumber` API docs I could find.

Comment (maxconway) [reply zilliqa/src/api/types/eth.rs:104]: Just for ZQ1 backwards compatibility

Comment (maxconway) [reply zilliqa/src/api/types/eth.rs:104]: Fixed here: https://github.com/Zilliqa/zq2/pull/1789

**Pull request merged**